datahub-web-react/src/app/domain/DomainSearch.tsx

```diff
@@ -102,38 +109,47 @@ function DomainSearch() {
                     entityRegistry={entityRegistry}
                     onFocus={() => setIsSearchBarFocused(true)}
                 />
-                {isSearchBarFocused && searchResults && !!searchResults.length && (
-                    <ResultsWrapper>
-                        {searchResults.map((result) => {
-                            return (
-                                <SearchResult
-                                    to={entityRegistry.getEntityUrl(result.entity.type, result.entity.urn)}
-                                    onClick={() => setIsSearchBarFocused(false)}
-                                >
-                                    <IconWrapper>
-                                        {result.entity.type === EntityType.Domain ? (
-                                            <DomainIcon
-                                                style={{
-                                                    fontSize: 16,
-                                                    color: '#BFBFBF',
-                                                }}
-                                            />
-                                        ) : (
-                                            entityRegistry.getIcon(result.entity.type, 12, IconStyleType.ACCENT)
-                                        )}
-                                    </IconWrapper>
-                                    <div>
-                                        <ParentEntities
-                                            parentEntities={getParentDomains(result.entity, entityRegistry)}
-                                        />
-                                        <Highlight matchStyle={highlightMatchStyle} search={query}>
-                                            {entityRegistry.getDisplayName(result.entity.type, result.entity)}
-                                        </Highlight>
-                                    </div>
-                                </SearchResult>
-                            );
-                        })}
-                    </ResultsWrapper>
+                {loading && (
+                    <LoadingWrapper>
+                        <LoadingOutlined />
+                    </LoadingWrapper>
+                )}
+                {!loading && (
+                    <>
+                        {isSearchBarFocused && searchResults && !!searchResults.length && (
+                            <ResultsWrapper>
+                                {searchResults.map((result) => {
+                                    return (
+                                        <SearchResult
```

Comment:
this logic could use some refactoring... do you mind perhaps breaking this up into another component?

Reply:
Sure

Reply:
i know difficult to switch back to this - from here on will keep things streamlined :)

Reply:
Np It's Done